app/assets/javascripts/app/views/location_stream.js

```diff
@@ -1,7 +1,49 @@
 // @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
 
 app.views.LocationStream = app.views.Content.extend({
-  templateName: "status-message-location"
+  events: {
+    "click .near-from": "toggleMap"
+  },
+  templateName: "status-message-location",
+
+  toggleMap: function () {
+    if (gon.appConfig.map.enabled){
+      var mapContainer = this.$el.find(".mapContainer");
+
+      if (mapContainer.hasClass("empty")) {
+        var location = this.model.get("location");
+        mapContainer.css("height", "150px");
+
+        if (location.lat) {
+
+          // If the mapbox option is enabled in the diaspora.yml, the mapbox tiles with the podmin's credentials are used.
+          // If mapbox is not enabled the Maptiles from the Heidelberg University are used, which don't need credentials.
+          var mapsource = gon.appConfig.map.mapbox.enabled ? gon.appConfig.map.mapbox : "";
+          var tileLayerSource = mapsource ? "https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}"
+                                          : "http://korona.geog.uni-heidelberg.de/tiles/roads/x={x}&y={y}&z={z}";
+          var tileAttribution = mapsource ? "<a href='http://creativecommons.org/licenses/by-sa/2.0/''>CC-BY-SA</a>, " +
+                                            "Imagery © <a href='https://www.mapbox.com'>Mapbox</a>"
+                                          : "rendering <a href='http://giscience.uni-hd.de/'>" +
+                                            "GIScience Research Group @ Heidelberg University</a>";
+          var attribution = "Map data &copy; <a href='http://openstreetmap.org'>OpenStreetMap</a> contributors, " +
+                            tileAttribution;
+
+          var map = L.map(mapContainer[0]).setView([location.lat, location.lng], 14);
+          L.tileLayer(tileLayerSource, {
+            id: mapsource.id,
```

Comment:
Here you call `mapsource.id`. What's `"".id` supposed to return?

So yes, a `if/else` around everything different here would probably a bit cleaner and allow for example to not pass these attributes in the first place if they're not needed instead of passing undefined.